File: benchmarks/official/suites/sieve/sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

int main(void) {
    int64_t limit = 100000;
    int64_t *sieve = (int64_t *)malloc((limit + 1) * sizeof(int64_t));
    for (int64_t i = 0; i <= limit; i++) {
        sieve[i] = 1;
    }

    for (int64_t i = 2; (i * i) <= limit; i++) {
        if (sieve[i] == 1) {
            for (int64_t j = i * i; j <= limit; j += i) {
                sieve[j] = 0;
            }
        }
    }

    int64_t count = 0;
    for (int64_t i = 2; i <= limit; i++) {
        if (sieve[i] == 1) {
            count++;
        }
    }

    printf("%lld\n", count);
    free(sieve);
    return 0;
}

File: benchmarks/official/suites/sieve/sieve.rs
fn main() {
    let limit: usize = 100_000;
    let mut sieve = vec![1i64; limit + 1];

    let mut i = 2;
    while (i * i) <= limit {
        if sieve[i] == 1 {
            let mut j = i * i;
            while j <= limit {
                sieve[j] = 0;
                j += i;
            }
        }
        i += 1;
    }

    let mut count = 0;
    for i in 2..=limit {
        if sieve[i] == 1 {
            count += 1;
        }
    }
    println!("{count}");
}
